src/InstrumentationEngine/ProfilerManager.h

```diff
@@ -368,7 +373,7 @@ namespace MicrosoftInstrumentationEngine
 
             CComPtr<TInterfaceType> pCallback;
 
-            shared_ptr<CProfilerCallbackHolder> pProfilerCallbackHolder = atomic_load(&m_profilerCallbackHolder);
+            CProfilerCallbackHolder* pProfilerCallbackHolder = InterlockedCompareExchangePointer(&m_profilerCallbackHolder, nullptr, nullptr);
```

Comment:
The function here does not seem to require the cast on the parameter or the return value while other invocations do.

Reply:
Right, I wasn't sure why the compiler was ok with this. I'll add the explicit static_cast.